src/main/java/org/wildfly/security/credential/KeyPairCredential.java

```diff
@@ -63,4 +63,9 @@ public KeyPair getKeyPair() {
     public String getAlgorithm() {
         return keyPair.getPublic().getAlgorithm();
     }
+
+    public KeyPairCredential clone() {
+        return this;
```

Comment:
Note that several of the JDK PrivateKey types don't override the destroy() method, so maybe that check could come earlier.

Also note that the result of reflection examination can be stashed in a ClassValue as a ClassValue<UnaryOperator<? extends Key>>.  This way you don't have to run through the same detection logic for the same class over and over.